samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -314,11 +314,11 @@ static int cmd_sock_connect(const struct shell *shell, size_t argc, char **argv)
 			arg_secure = true;
 			break;
 		case 'T': /* Security tag */
-			arg_sec_tag = atoi(optarg);
-			if (arg_sec_tag < 0) {
+			arg_sec_tag = shell_strtoul(optarg, 10, &err);
```

Comment:
I was unaware of `shell_strtoul()` with error checks, nice. It seems, that the function does not set the `err` reference to zero in the success case. Now `err` happens to be zero, but just wondering if it should anyway be initialized to zero before the function call, just in case.

Reply:
I don't know if I like the pattern of always setting `err = 0` first. If `err` is not zero the code will always return, and the extra init should not be needed.

We should use `shell_strtol()` and `shell_strtoul()` instead of `atoi()` in all shell code (~ 80 places), and it's good to determine this pattern first.

Reply:
I don't like it either, I would have preferred that `shell_strto(u)l()` would have done it. If `err` is not explicitly initialized before the function call, you always need to check the code to make sure it's zero before the function is called. Even though it works now, somebody may later modify one of the cases so, that `err` may not always be zero. There can be cases, where `err` would be set to a non-zero value, but it would not be considered an error and processing of the parameters would be continued. I don't like that we have to make assumptions about how unrelated code works. @trantanen?

Reply:
> I don't know if I like the pattern of always setting `err = 0` first. If `err` is not zero the code will always return, and the extra init should not be needed.

So the code will return when an integer is successfully parsed in cases when `err` happens to be non-zero in the memory before calling the function. I don't think this is what we want or what am I missing?

Reply:
So is there a better alternative than simply setting `err = 0` before `shell_strto(u)l()` calls?
An own wrapper function could do it but I'd prefer `err = 0`.

Reply:
Ok, I'll do `err = 0` for now.